### DISCUSSION TITLE
Google Apps Script project files mis-identified as Gosu, but they're vanilla JS

### DISCUSSION THREAD

**Discussion by rclayton-neon** [github-linguist/linguist#6667] [Classification]
## Describe the bug
.gs files may not be Gosu.  I have a Google Apps Script (Google Workspace apps, Google Spreadsheet apps, etc) app that Github is misidentifying the language(s) on, they referred me to you all.

[this](https://github.com/jpcamara/Textmate-Gosu-Bundle/blob/c0e36a01ba2c82d87fd6623481fbb57a5055b376/Gosu.tmbundle/Syntaxes/Gosu.tmLanguage#L7) seems to be the offending line, but I'm not sure of a fix beyond that since I'm not familiar with your functionality.

### Expected behaviour
The file in question is all vanilla javascript (all Google Workspace apps ending in .gs are vanilla javascript, to my knowledge).  [Google's Guide link](https://developers.google.com/apps-script/guides/projects)

### Related discussion
referred by Github support ticket #2509380

### Additional notes
N/A

**Comment by lildude** [comment]
The reason this is happening is because the `.gs` extension is associated with multiple languages, including JavaScript, however the heuristic doesn't have a specific entry for JavaScript:
https://github.com/github-linguist/linguist/blob/ee9119e70111d46842788b2bc1666ec1d1ea9e2a/lib/linguist/heuristics.yml#L307-L314

... so either your files are matching the Gosu entry or are falling through to the classifier which is making a guess based on the samples we have.

The fix is to submit a PR to extend the heuristics to detect all instances of JavaScript `.gs` files correctly or as more samples in the hope this will better train the classifier, but this isn't likely to be foolproof or particularly reliable.

Until then, you can use an [override](https://github.com/github-linguist/linguist/blob/master/docs/overrides.md) to force the language.
